### DISCUSSION TITLE
Dreamhost discontinuing Passenger support

### DISCUSSION THREAD

**Discussion by webmahi** [jjlin/vaultwarden-shared-hosting#4] [Q&A]
Recently got the below from Dreamhost. I have no experience with Passenger. Can anyone help with what needs to be done to keep our Dreamhost VPS hosted Vaultwardens safely operational?

Thanks

> We've recently made some changes to our Hosting platform that will impact websites on your account using Passenger.
> 
> After July 30th, we will no longer be able to offer Passenger on VPS or Dedicated Server plans. As such, sites running applications currently relying on Passenger (eg. React, Node, Flask, Ruby, etc.) will need to be set up with a [Proxy Server](https://help.dreamhost.com/hc/en-us/articles/217955787-Proxy-Server). 
> 
> **Next Steps**
> 
> In order to keep those site(s) active, we strongly encourage you to look into setting up your Passenger sites using a Proxy Server, as well as setting up [Gunicorn and Linger](https://help.dreamhost.com/hc/en-us/articles/23971547819412-Using-linger-with-Gunicorn) to allow for persistent processes.
> 
> Please note after July 30th, your sites running Passenger will no longer function correctly without transitioning to a setup like the ones suggested above. Please plan accordingly.

**Comment by jjlin** [comment]
I don't have access to DreamHost anymore, but given that they seem to be offering this "Proxy Server" service now, you can presumably just use that to proxy requests to Vaultwarden on whichever port you're running it on, so it should actually be much simpler. Proxying requests was the whole point of the `passenger_wsgi.py` script anyway.
